### PR TITLE
feat: row span

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description
**rowSpan**
   * calculate rowSpan index or position when focus, selection, virtual scrolling
   * update rowSpan when append or remove row
   * store rowSpan data in `rowSpanMap`, not `_attribute`
   * rowSpanCell component
   * getRowSpanData() API
   * storybook(`visual test - selection, focus, virtual scrolling`)
   * cypress test(`API test`)

###TODO
   * next cell index related rowSpan in keyboard selection
   * multi sort when calls unsort API

###DEMO
   * http://10.78.9.214:6006/?path=/story/rowspan--rowspan

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/helper/common.ts [hunk 1]
```diff
@@ -251,15 +251,15 @@ export function isString(value: unknown): value is string {
...
  */
-export function isEmpty(obj: object | any[]) {
+export function isEmpty(obj: any) {
   if (
     isNull(obj) ||
     isUndefined(obj) ||
-    (!isUndefined((obj as any[]).length) && (obj as any[]).length === 0) ||
+    (!isUndefined(obj.length) && obj.length === 0) ||
     Object.keys(obj).length === 0
   ) {
```
Comment (jung-han) [inline src/helper/common.ts]: 그냥 
```js
return isNull(obj) ||
    isUndefined(obj) ||
    (!isUndefined(obj.length) && obj.length === 0) ||
    Object.keys(obj).length === 0;
```
을 해도 되겠네요

File: src/helper/rowSpan.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+ * get max rowSpan count in all columns that have rowSpan
+ */
+export function getMaxRowSpanCount(rowIndex: number, data: Row[]) {
+  const { rowSpanMap } = data[rowIndex];
+
+  if (isEmpty(rowSpanMap)) {
+    return 0;
+  }
+
+  const maxRowSpanCount = Object.keys(rowSpanMap).reduce(
```
Comment (jung-han) [inline src/helper/rowSpan.ts]: 여기도 따로 변수에 할당하지 않고 return 해도 될 것 같습니다.

File: src/helper/common.ts (hunk 1 above)
Comment (js87zz) [reply src/helper/common.ts]: 넵 수정하겠습니다

File: src/helper/rowSpan.ts (hunk 2 above)
Comment (js87zz) [reply src/helper/rowSpan.ts]: 넵 수정하겠습니다.

File: src/helper/rowSpan.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  data: Data
+): Range {
+  const sortedColRange = getSortedRange(colRange);
+  const endRowIndex = rowRange[1];
+
+  let startRowIndex = rowRange[0];
+
+  /* if start row index is different from focused index ,
+   * change start row index to focused row index for getting proper row range
+   */
```
Comment (dongwoo-kim) [inline src/helper/rowSpan.ts]: 요기 주석 스타일이 약간 애매한데.. 그냥 `//` 두 줄 써도 되지 않을까요? ㅎ

Comment (js87zz) [reply src/helper/rowSpan.ts]: 넵 수정하겠습니다

File: src/helper/rowSpan.ts [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+  }
+
+  const maxRowSpanCount = Object.keys(rowSpanMap).reduce(
+    (acc, columnName) => Math.max(acc, rowSpanMap[columnName].spanCount),
+    0
+  );
+  return maxRowSpanCount;
+}
+
+export function enableRowSpan(data: Data) {
```
Comment (jung-han) [inline src/helper/rowSpan.ts]: columnName만을 매개변수로 넘겨도 되지 않을까요?

Comment (js87zz) [reply src/helper/rowSpan.ts]: 호출할 때마다 `enableRowSpan(data.sortOptions.columnName)`로 하는 게 번거로워서 내부적으로 처리하는 식으로 했는데 변경하는게 나을까요?

**Pull request merged**